dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/PropertiesHelper.java

```diff
@@ -0,0 +1,109 @@
+        Properties fileProperties = readProperties();
+        Properties properties = replacePrefix(fileProperties);
+        Properties augmentedProperties = augmentProperties(properties, commandLine);
+        commandLine.setDefaultValueProvider(new CommandLine.PropertiesDefaultProvider(augmentedProperties));
+    }
+
+    private static Properties augmentProperties(Properties properties, final CommandLine commandLine) {
+        List<String> commonArgumentList = commonArgumentList(new ArrayList<>(), commandLine.getSubcommands(), commandLine.getCommandName());
+        commonArgumentList.forEach(key -> {
+            if (!properties.contains(key)) {
+                String[] parts = key.split(Pattern.quote("."));
+                String arg = parts[parts.length - 1];
+                if (COMMON_ARGUMENTS.contains(arg) && properties.containsKey(COMMON_PREFIX + arg)) {
+                    String value = properties.getProperty(COMMON_PREFIX + arg);
+                    properties.put(key, value);
+                }
+            }
+        });
+        return properties;
+    }
+
+    private static List<String> commonArgumentList(List<String> list, Map<String, CommandLine> commandLines, String parentPrefix) {
+        commandLines.forEach((name, subCommand) -> {
+            subCommand.getCommandSpec().args().stream()
+                    .filter(arg -> arg instanceof CommandLine.Model.OptionSpec)
+                    .map(arg -> (CommandLine.Model.OptionSpec) arg)
+                    .filter(arg -> COMMON_ARGUMENTS.contains(arg.longestName().replace("--", "")))
+                    .map(arg -> arg.longestName().replace("--", ""))
+                    .forEach(arg -> list.add(generateParameter(parentPrefix, subCommand.getCommandName(), arg)));
+
+            list.addAll(commonArgumentList(new ArrayList<>(), subCommand.getSubcommands(), generateParameter(parentPrefix, name)));
+        });
+        return list;
+    }
+
+    private static String generateParameter(String... prefix) {
+        return Arrays.asList(prefix).stream()
+                .filter(s -> s != null)
+                .collect(Collectors.joining("."));
+    }
+
+    private static Properties replacePrefix(Properties properties) {
+        Properties result = new Properties();
+        properties.forEach((key, value) -> {
+            if (key.toString().startsWith(PROPERTY_PREFIX)) {
+                result.put(key.toString().replace(PROPERTY_PREFIX, COMMAND_PREFIX), value);
+            }
+        });
+        return result;
+    }
+
+    private static Properties readProperties() {
+        File defaultsFile = new File(APPLICATION_PROPERTIES_FILE);
+        Properties properties = new Properties();
+        if (defaultsFile.exists()) {
+            try (FileInputStream fis = new FileInputStream(defaultsFile)) {
+                properties.load(fis);
+            } catch (IOException e) {
+                LOG.error("Error", e);
```

Comment:
I am probably nitpicking a bit, but ... maybe wouldn't it be better to also log the exception message? I.e.: 

```
LOG.error("Error reading defaults file: {}", e.getMessage, e); 
```

Reply:
Thanks @orpiske